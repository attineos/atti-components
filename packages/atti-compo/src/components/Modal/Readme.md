A Modal :

```js
initialState = { isOpen: false }
;<div>
  <Button onClick={() => setState({ isOpen: true })}>Open Modal</Button>
  <Modal
    isOpened={state.isOpen}
    onClose={() => setState({ isOpen: false })}
    title="This is the modal's title"
  >
    <Text>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce quis felis quis lorem lobortis
      lobortis vel non tortor. Aliquam tempor ullamcorper dolor vel ultricies. Etiam molestie ut mi
      ut posuere. Cras ut purus aliquam, rhoncus lacus quis, tristique mi. Nunc at arcu et nisi
      rutrum tristique. Aenean ullamcorper cursus rutrum. Praesent id maximus mauris. Sed eros
      justo, volutpat a placerat lobortis, posuere nec neque. Ut condimentum placerat lacinia. Nam
      vitae lorem risus.{' '}
    </Text>
    <br />
    <Text>
      Etiam finibus congue cursus. Aliquam vel metus at leo vestibulum rhoncus ac nec purus. Sed
      feugiat dolor id libero pharetra lobortis. Nullam vulputate pharetra odio ac malesuada. Nulla
      consequat sem vitae enim cursus, eget feugiat lectus tincidunt. Quisque at placerat lectus.
      Fusce vel metus ac tellus rhoncus varius.
    </Text>
  </Modal>
</div>
```
