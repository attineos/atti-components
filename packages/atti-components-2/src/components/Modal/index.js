import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Header3 } from '../Typographies'
import {
  CloseButton,
  ContentContainer,
  ModalBackground,
  ModalContainer,
  TitleContainer,
} from './styles'

/**
 * A Modal.
 *
 */
const Modal = ({ children, className, isOpened, onClose, title }) => {
  const [opened, setOpened] = useState(isOpened)

  useEffect(() => {
    setOpened(isOpened)
  }, [isOpened])

  const closeModal = () => {
    onClose && onClose()
    setOpened(false)
  }

  return (
    opened && (
      <ModalBackground className={className} onClick={closeModal}>
        <ModalContainer onClick={e => e.stopPropagation()}>
          <TitleContainer>
            <Header3>{title}</Header3>
            <CloseButton type="button" onClick={closeModal} aria-label="Close Modal">
              <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
                <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
                <path d="M0 0h24v24H0z" fill="none" />
              </svg>
            </CloseButton>
          </TitleContainer>
          {children && <ContentContainer>{children}</ContentContainer>}
        </ModalContainer>
      </ModalBackground>
    )
  )
}

Modal.defaultProps = {
  children: '',
  className: '',
  isOpened: false,
  onClose: null,
}

Modal.propTypes = {
  /**
   * The content of the Modal.
   */
  children: PropTypes.any,
  /**
   * The classes of the Modal.
   */
  className: PropTypes.string,
  /**
   * Is the modal opened or not.
   */
  isOpened: PropTypes.bool,
  /**
   * Function executed when the Modal is closed.
   */
  onClose: PropTypes.func,
  /**
   * The title of the Modal.
   */
  title: PropTypes.string.isRequired,
}

/** @component */
export default styled(React.memo(Modal))``
