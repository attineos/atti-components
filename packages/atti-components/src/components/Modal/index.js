import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Header3 } from '../Typographies'
import {
  CloseButton,
  ContentContainer,
  ModalBackground,
  ModalContainer,
  TitleContainer,
} from './styles'

/**
 * A Modal.
 *
 */
const Modal = ({ children, isOpened, onClose, title, ...rest }) => {
  const [opened, setOpened] = useState(isOpened)

  useEffect(() => {
    setOpened(isOpened)
  }, [isOpened])

  const closeModal = () => {
    onClose && onClose()
    setOpened(false)
  }

  return (
    opened && (
      <ModalBackground onClick={closeModal} {...rest}>
        <ModalContainer onClick={e => e.stopPropagation()}>
          <TitleContainer>
            <Header3>{title}</Header3>
            <CloseButton type="button" onClick={closeModal} aria-label="Close Modal">
              <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
                <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
                <path d="M0 0h24v24H0z" fill="none" />
              </svg>
            </CloseButton>
          </TitleContainer>
          {children && <ContentContainer>{children}</ContentContainer>}
        </ModalContainer>
      </ModalBackground>
    )
  )
}

Modal.defaultProps = {
  children: '',
  isOpened: false,
  onClose: null,
}

Modal.propTypes = {
  /**
   * The content of the Modal.
   */
  children: PropTypes.any,
  /**
   * Is the modal opened or not.
   */
  isOpened: PropTypes.bool,
  /**
   * Function executed when the Modal is closed.
   */
  onClose: PropTypes.func,
  /**
   * The title of the Modal.
   */
  title: PropTypes.string.isRequired,
}

/** @component */
export default styled(Modal)``
