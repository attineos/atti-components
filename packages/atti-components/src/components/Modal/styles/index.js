import CloseButton from './CloseButton'
import ContentContainer from './ContentContainer'
import ModalBackground from './ModalBackground'
import ModalContainer from './ModalContainer'
import TitleContainer from './TitleContainer'

export { CloseButton, ContentContainer, ModalBackground, ModalContainer, TitleContainer }
