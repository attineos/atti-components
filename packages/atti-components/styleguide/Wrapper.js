import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
