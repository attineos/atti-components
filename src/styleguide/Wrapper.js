import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
