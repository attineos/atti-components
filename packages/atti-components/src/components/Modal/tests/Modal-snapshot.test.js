/* This file was generated automatically, please do not write in it. */
import React from 'react'
import Modal from '..'
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer'

const mock = () => {}

it('Modal renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <Modal className="Atticomponents" isOpened={true} onClose={mock} title="Atticomponents">
          Atticomponents
        </Modal>
      </Wrapper>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
