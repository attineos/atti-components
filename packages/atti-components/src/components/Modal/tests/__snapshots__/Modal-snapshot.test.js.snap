// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<span
  className="atti-box"
>
  <span
    className="atti-box"
  >
    <div
      className="Atticomponents Modal-sc-1q6w6or-0 bbGikj ModalBackground-stke3x-0 hnRWcO"
      onClick={[Function]}
    >
      <div
        className="ModalContainer-g0jgsc-0 bOiHpR"
        onClick={[Function]}
      >
        <div
          className="TitleContainer-sc-1wmifrj-0 jbdmOp"
        >
          <h3
            className="Typography-csduj4-0-h3 Header3-q478ae-0 atti-r-css  lfYxUS"
          >
            Atticomponents
          </h3>
          <button
            aria-label="Close Modal"
            className="CloseButton-sc-1ujuxrw-0 klRTOh"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
        </div>
        <aside
          className="ContentContainer-sc-1i6cggq-0 eHwzTP"
        >
          Atticomponents
        </aside>
      </div>
    </div>
  </span>
</span>
`;
